ambry-cloud/src/main/java/com/github/ambry/cloud/CloudDestination.java

```diff
@@ -150,4 +152,41 @@ boolean retrieveTokens(String partitionPath, String tokenFileName, OutputStream
    * @return {@link CloudContainerCompactor} object that would do container compaction for cloud.
    */
   CloudContainerCompactor getContainerCompactor();
+
+  /**
+   * To be consistent with MockServer. We need the following four error injection APIs.
+   * Different from MockServer, we use {@link StoreErrorCodes} instead of {@link ServerErrorCode} as the error code.
+   */
+
+  /**
+   * Set the error for each request from this point onwards that affects subsequent requests sent to this node
+   * (until/unless the next set or reset error method is invoked).
+   * Each request from the list is used exactly once and in order. So, if the list contains {No_Error, Unknown_Error,
+   * Disk_Error}, then the first, second and third requests would receive No_Error,
+   * Unknown_Error and Disk_Error respectively. Once the errors are exhausted, the default No_Error is assumed for
+   * all further requests until the next call to this method.
+   * @param serverErrors the list of errors that affects subsequent PutRequests.
+   */
+  void setServerErrors(List<StoreErrorCodes> serverErrors);
+
+  /**
+   * Set the error to be set in the responses for all requests from this point onwards (until/unless another set or
+   * reset method for errors is invoked).
+   * @param serverError the error to set from this point onwards.
+   */
+  void setServerErrorForAllRequests(StoreErrorCodes serverError);
+
+  /**
+   * Clear the error for subsequent requests. That is all responses from this point onwards will be successful
+   * until/unless another set error method is invoked.
+   */
+  void resetServerErrors();
+
+  /**
+   * Sets up this {@link CloudDestination} such that it returns the given {@code errorCode} for the given {@code blobId} for
+   * get, ttl update and delete (not put).
+   * @param blobId the blob id for which the error code must apply
+   * @param errorCode the error code to apply
+   */
+  void setErrorCodeForBlob(String blobId, StoreErrorCodes errorCode);
```

Comment:
I don't recommend include these methods in the interface. The interface should be a clean interface that only contains valid methods. If we want to inject errors, we should do this in the mock implementation, not in the interface.

Reply:
Make sense. Removed it.